tedana/decomposition/_utils.py

```diff
@@ -40,3 +41,51 @@ def eimask(dd, ees=None):
         imask[np.logical_and(m > lthr, m < hthr), ee] = True
 
     return imask
+
+
+def dwtmat(mmix):
+    """
+    Wavelet transform data.
+
+    Parameters
+    ----------
+    mmix : :obj:`numpy.ndarray`
+        Data to wavelet transform.
+
+    Returns
+    -------
+    mmix_wt : :obj:`numpy.ndarray`
+        Wavelet-transformed data.
+    cAlen : :obj:`int`
+        Index of some kind?
+    """
+    llt = len(np.hstack(pywt.dwt(mmix[0], 'db2')))
+    mmix_wt = np.zeros([mmix.shape[0], llt])
+    for ii in range(mmix_wt.shape[0]):
+        wtx = pywt.dwt(mmix[ii], 'db2')
```

Comment:
From the [pywavelets documentation](http://pywavelets.readthedocs.io/en/latest/regression/dwt-idwt.html#discrete-wavelet-transform), it seems that `wtx` would be the approximation coefficients for the specified row of `mmix`. Then `cAlen` would be the length of the first entry of `wtx` in the last for-loop.

If you agree with that assessment, could we improve these variable names ?

Reply:
I think it's the approximation and detail coefficients, not that I'm sure what those are. `cAlen` is the number of coefficients (i.e., `len` of `cA`). Oh, what about `n_coefs_approx` instead of `cAlen`, `coefs_tup` instead of `wtx`, `n_coefs_total` instead of `llt`, `mmix_coefs` instead of `mmix_wt`?

What does `mmix` stand for again?